pkg/controllers/netpol/nftables.go

```diff
@@ -356,14 +357,24 @@ func genIPSetFromEgressRule(rule EgressRule) string {
 }
 
 func getOrCreateSet(unfiltered []string) string {
-	ips := make([]string, 0)
+	nets := make([]string, 0)
 	for _, ip := range unfiltered {
 		if len(ip) > 1 {
-			ips = append(ips, ip)
+			if addr := net.ParseIP(ip); addr != nil { // if this is a plain address (non-cidr notation, add /32)
+				ip = ip + "/32"
+			}
+			nets = append(nets, ip)
 		}
 	}
-	sort.Strings(ips)
-	ipStr := strings.Join(ips, ",")
+	merged, err := MergeCIDRs(nets)
+	if err != nil {
+		// the thing is: "getOrCreateSet" is not really allowed to fail, so if merge fails, just log it and use the unmerged list
+		glog.Errorf("merging of cidrs failed for set: %v", nets, err)
```

Comment:
My only concern here is whether it's better to bail harder here, so we don't risk feeding overlaps to nftables under any circumstances? I guess the user-facing (well, node-facing) error would be much the same regardless?